src/ImageSharp/Formats/Png/Zlib/DeflaterHuffman.cs

```diff
@@ -122,12 +74,77 @@ public DeflaterHuffman(MemoryAllocator memoryAllocator)
             this.pinnedLiteralBuffer = (short*)this.literalBufferHandle.Pointer;
         }
 
+        // See RFC 1951 3.2.6
+        // Literal codes
+        private static short[] StaticLCodes => new short[]
```

Comment:
I'm confused, why is this a property returning an array, instead of a statically initialized array?
This will create a new array every time it's used, and copy the data there.
Am I missing something obvious or is this just an oversight? 🤔

Reply:
That's a mistake caused by copy/paste.

Btw... I had a bit of a read and FindLongestMatch should definitely be the focus. Looking at a well known implementation we're doing far too much. I had a quick hack and got around 80% of the tests passing but soon hit a wall.

https://github.com/zlib-ng/zlib-ng/blob/ee1c9bc0be715d788513ac3cbfe6559f91c6b1d3/match_p.h#L49

Reply:
> That's a mistake caused by copy/paste.

Though it's a static property so it shouldn't be requesting a new array each time.

Reply:
Well no, because it's just a getter there. You don't have a static _field_ that is initialized at some point and that you then use to get a reference to that array, you just have a getter that returns a new array every time. That's exactly the same as this:
```csharp
public static short[] StaticLCodes_Get() => new short[] { ... };
```

Reply:
Sorry, was getting mixed up with the `ReadonlySpan` optimization https://github.com/dotnet/roslyn/pull/24621/files